km/km_coredump.c

```diff
@@ -959,6 +962,51 @@ int km_dump_core(char* core_path,
       }
    }
 
+   if (dumptype == KM_DO_SNAP) {
+      int cfd;   // conf file fd
+      char cpath[MAXPATHLEN];
+      sprintf(cpath, "%s.conf", core_path);
+      if ((cfd = open(cpath, O_RDWR | O_CREAT | O_TRUNC, 0600)) < 0) {
+         km_warn("Cannot create %s '%s'", "snapshot config", cpath);
+         return errno;
```

Comment:
You should "goto out;" here instead.
That will free memory and close the open core file fd.
And, we should probably delete the snapshot if we are failing the snapshot (that is not there currently).